addons/casings/config.cpp

```diff
@@ -13,5 +13,13 @@ class CfgPatches {
         VERSION_CONFIG;
     };
 };
+class CfgAmmo {
+    class GrenadeBase; // Reference the base class
 
+    class G_40mm_HE: GrenadeBase {
+        // Override the specific parameters here
+        cartridge = "FxCartridge_40mm";
```

Comment:
This class doesn't exist. This *will* cause issues at some point.

Reply:
maybe we add a new config
`QGVAR(modelPath) = x.p3d`


```
_modelPath = getText (configFile >> "CfgAmmo" >> _ammo >> QGVAR(modelPath));

if (_modelPath == "") then {
    private _cartridge = getText (configFile >> "CfgAmmo" >> _ammo >> "cartridge");
// use cartridge lookup switch...
```
also, we could move the cartridge->modelPath to be config based instead of the `switch`?

I'm happy to do the work if this sounds ok

Reply:
I was doing that exact thing but got ADHD'd.

Also we'd need handling for firing UBGLs in this specific case, as most of those shouldn't drop casings until reload.

Reply:
could handle shells on reload
```
["CAManBase", "Reloaded", {
    params ["_unit", "_weapon", "_muzzle", "_newMagazine", "_oldMagazine"];
    if (_unit != ACE_player) exitWith {};
    if (isNil "_oldMagazine") exitWith {};
    // if (!_muzzle isKindOf GL) exitWith {};
    _oldMagazine params ["_mag", "_ammo"];
    if (_ammo != 0) exitWith {};

    systemChat "making shell"; 
    //global event?

}] call CBA_fnc_addClassEventHandler;
```

Reply:
At that point, might as well have a mag drop component for spent reloads (this is why dropped work on this)

Reply:
> Magazines are not typically dropped unless things are really dire so I personally don't think you need to worry about them

Very true, but in Arma they vanish into thin air once you reload your gun and the magazine loaded in the gun is empty (unless you are running the Green Ammo mod). If we were to take grenade casings after reload into account, why not empty magazines? Empty magazines would either need to be returned into the player's inventory or dropped onto the ground.

---

Going down this route is going to open a can of worms imo. I'm with @LinkIsGrim - this needs to be thought out better and done properly.